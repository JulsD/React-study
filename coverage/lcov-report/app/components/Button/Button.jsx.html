<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/Button/Button.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">app/components/Button</a> Button.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Link } from 'react-router-dom'
&nbsp;
import styles from './button.css';
&nbsp;
class Button extends React.Component {
  constructor(props) {
    super(props);
  }
&nbsp;
  render() {
&nbsp;
    const btnStyleMap = {
      'primary': styles.primary,
      'default': styles.default,
      'informative': styles.informative,
      'not-active': styles['not-active']
    }
    
    if(this.props.type == 'button' || this.props.type == 'submit') {
      return (
          &lt;button type={this.props.type} className={`${btnStyleMap[this.props.kind]}`}&gt;
            {this.props.children}
          &lt;/button&gt;
      );
    } else if (this.props.type == 'link'){
      return (
          &lt;Link to={this.props.to} className={`${btnStyleMap[this.props.kind]}`}&gt;
            {this.props.children}
          &lt;/Link&gt;
      );
    } else if (this.props.type == 'a'){
      return (
          &lt;a to={this.props.href} className={`${btnStyleMap[this.props.kind]}`}&gt;
            {this.props.children}
          &lt;/a&gt;
      );
    } else if (this.props.type == 'label'){
      return (
          &lt;label htmlFor={this.props.for} className={`${btnStyleMap[this.props.kind]}`}&gt;
            {this.props.children}
          &lt;/label&gt;
      );
    } else {
      return (
          &lt;div className={`${btnStyleMap[this.props.kind]}`}&gt;
            {this.props.children}
          &lt;/div&gt;
      );
    }
  }
&nbsp;
}
&nbsp;
export default Button;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 09 2017 00:12:39 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
