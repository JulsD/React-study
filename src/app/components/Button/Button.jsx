import React from 'react';
import { Link } from 'react-router-dom'

import styles from './button.css';

class Button extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {

    const btnStyleMap = {
      'primary': styles.primary,
      'default': styles.default,
      'informative': styles.informative,
      'not-active': styles['not-active']
    }
    
    if(this.props.type == 'button' || this.props.type == 'submit') {
      return (
          <button type={this.props.type} className={`${btnStyleMap[this.props.kind]} ${styles.root}`}>
            {this.props.children}
          </button>
      );
    } else if (this.props.type == 'link'){
      return (
          <Link to={this.props.to} className={`${btnStyleMap[this.props.kind]} ${styles.root}`}>
            {this.props.children}
          </Link>
      );
    } else if (this.props.type == 'a'){
      return (
          <a to={this.props.href} className={`${btnStyleMap[this.props.kind]} ${styles.root}`}>
            {this.props.children}
          </a>
      );
    } else if (this.props.type == 'label'){
      return (
          <label htmlFor={this.props.for} className={`${btnStyleMap[this.props.kind]} ${styles.root}`}>
            {this.props.children}
          </label>
      );
    } else {
      return (
          <div className={`${btnStyleMap[this.props.kind]} ${styles.root}`}>
            {this.props.children}
          </div>
      );
    }
  }

}

export default Button;
