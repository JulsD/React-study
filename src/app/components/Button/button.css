.root {
  padding: 5px 10px;
  text-transform: uppercase;
  border: none;
  outline: none;
  font-size: 13px;
  font-weight: 500;
  letter-spacing: -.2px;
  border-radius: 2px;
  cursor: pointer;
}
.primary {
  background-color: #f40612;
  color: #fff;
}
.default {
  background-color: #f3f3f3;
  color: #f40612;
}
.informative {
  padding: 3px;
  border: 1px solid black;
}
.not-active {
  background-color: #f3f3f3;
  color: #666;
}
