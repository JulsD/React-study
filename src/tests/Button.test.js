import React from 'react';
import Button from '../app/components/Button';

test('Button with type `button` should render a <button>', () => {
  const wrapper = shallow(
    <Button type='button'/>
  );
  expect(wrapper).toMatchSnapshot();
});

test('Button with type `link` should render a <Link>', () => {
  const wrapper = shallow(
    <Button to='href' type='link'/>
  );
  expect(wrapper).toMatchSnapshot();
});

test('Button with type `link` and `to` attr should render a <Link to="attr">', () => {
  const wrapper = shallow(
    <Button to='href' type='link'/>
  );
  const button = wrapper.find('Link');
  expect(button.prop('to')).toEqual('href');
});

test('Button with type `a` should render a <a>', () => {
  const wrapper = shallow(
    <Button to='href' type='a'/>
  );
  expect(wrapper).toMatchSnapshot();
});

test('Button with type `label` should render a <label>', () => {
  const wrapper = shallow(
    <Button htmlFor='id' type='label'/>
  );
  expect(wrapper).toMatchSnapshot();
});

test('Button without type should render a <button>', () => {
  const wrapper = shallow(
    <Button />
  );
  expect(wrapper).toMatchSnapshot();
});