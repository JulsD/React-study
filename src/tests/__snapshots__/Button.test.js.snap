// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button with type \`a\` should render a <a> 1`] = `
<a
  className="undefined"
/>
`;

exports[`Button with type \`button\` should render a <button> 1`] = `
<button
  className="undefined"
  type="button"
/>
`;

exports[`Button with type \`label\` should render a <label> 1`] = `
<label
  className="undefined"
/>
`;

exports[`Button with type \`link\` should render a <Link> 1`] = `
<Link
  className="undefined"
  replace={false}
  to="href"
/>
`;

exports[`Button without type should render a <button> 1`] = `
<div
  className="undefined"
/>
`;
